utils/build/docker/ruby/parametric/server.rb

```diff
@@ -36,515 +38,826 @@
 
 if Datadog::Core::Remote.active_remote
   # TODO: Remove this whole `if` condition if remote configuration is started by default.
-  raise "Remote Configuration worker already started! Remove this check and `Datadog::Core::Remote.active_remote.start` below." if Datadog::Core::Remote.active_remote.started?
+  if Datadog::Core::Remote.active_remote.started?
+    raise 'Remote Configuration worker already started! Remove this check and `Datadog::Core::Remote.active_remote.start` below.'
+  end
+
   Datadog::Core::Remote.active_remote.start
 end
 
+def otel_tracer
+  OpenTelemetry.tracer_provider.tracer('otel-tracer')
+end
+
+OTEL_SPAN_KIND = {
+  1 => :internal,
+  2 => :server,
+  3 => :client,
+  4 => :producer,
+  5 => :consumer
+}
+
 # Ensure output is always flushed, to prevent a forced shutdown from losing all logs.
 STDOUT.sync = true
 puts 'Loading server classes...'
 
-class ServerImpl < APMClient::Service
+DD_SPANS = {}
+OTEL_SPANS = {}
+DD_TRACES = {}
 
-  def crash(crash_args, _call)
-    STDOUT.puts "Crashing server..."
-    fork do
-      Process.kill('SEGV', Process.pid)
-    end
+class HeaderTuple
+  attr_accessor :key, :value
 
-    Process.wait2
-    CrashReturn.new
+  def initialize(key:, value:)
+    @key = key
+    @value = value
   end
+end
```

Comment:
We can go with a struct here and it will be similar

```ruby
HeaderTuple = Struct.new(:key, :value, keyword_init: true)
```

Reply:
Neat!